### PR TITLE
gitserver: make list of ignored git commands for recording configurable

### LINKED ISSUE #55059: git debuggability: make excluded git commands configurable via site config
/cc @sourcegraph/source

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #55059

This PR makes the list of git commands that should be ignored for recording configurable via the site config setting.

## Test plan

* Add an entry to the `gitRecorder.ignoredGitCommand` array.
* Perform an operation that will execute that command inside of `gitServer`, check redis for the value associated with key `recording-cmd` and it shouldn't be part of the result.

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM!
But you have some accidental (?) diff in `site.schema.json`, which fails the build, because it's not pretty enough :)

File: cmd/gitserver/shared/shared.go [hunk 1]
```diff
@@ -615,24 +615,34 @@ func methodSpecificUnaryInterceptor(method string, next grpc.UnaryServerIntercep
...
+	}
+
 	// we won't record any git commands with these commands since they are considered to be not destructive
-	ignoredGitCommands := map[string]struct{}{
-		"show":      {},
-		"rev-parse": {},
-		"log":       {},
-		"diff":      {},
-		"ls-tree":   {},
+	var ignoredGitCommandsMap = make(map[string]struct{}, len(ignoredGitCommands))
```
[the remark this answers is not in the document]
Comment (BolajiOlajide) [reply cmd/gitserver/shared/shared.go:641]: Oh! I didn't know about this. 
Thanks for the tip.

[the remark this answers is not in the document]
Comment (sashaostrikov) [reply cmd/gitserver/shared/shared.go:641]: yeah, golang doesn't have it, but Milan created it when he thought "enough with these maps!"

[the remark this answers is not in the document]
Comment (BolajiOlajide) [reply cmd/gitserver/shared/shared.go:641]: Super slick!

**Pull request merged**